credentials/alts/internal/handshaker/handshaker.go

```diff
@@ -198,6 +194,16 @@ func (h *altsHandshaker) ClientHandshake(ctx context.Context) (net.Conn, credent
 		return nil, nil, errors.New("only handshakers created using NewClientHandshaker can perform a client handshaker")
 	}
 
+	// TODO(matthewstevenson88): Change unit tests to use public APIs so
+	// that h.stream can unconditionally be set based on h.clientConn.
+	if h.stream == nil {
```

Comment:
Doesn't this have to be protected by a mutex? What happens if there are multiple concurrent calls to ClientHandshake?